scene/gui/control.cpp

```diff
@@ -200,20 +200,41 @@ void Control::reparent(Node *p_parent, bool p_keep_global_transform) {
 void Control::get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const {
 	ERR_READ_THREAD_GUARD;
 	Node::get_argument_options(p_function, p_idx, r_options);
+	if (!r_options->is_empty()) {
+		// If arguments exist, it means some other function was already matched.
+		return;
+	}
```

Comment:
Seems like the issue here is that `Control` is the only class which calls parent method before doing its own logic. I think this needs to be corrected instead and the above call (or rather `CanvasItem::get_argument_options` in master) should be moved to the end of this method.

Reply:
The comment explains why is it here. `get_argument_options()` can return arguments only for one method/argument, assuming it's implemented correctly, so if base class filled the list, there is no way to match another method.

Reply:
CC @Mickeon who worked on these completion methods recently.

Reply:
> The comment explains why is it here. `get_argument_options()` can return arguments only for one method/argument, assuming it's implemented correctly, so if base class filled the list, there is no way to match another method.

I don't understand what you mean. How is Control/CanvasItem different from the other inheritance chains that use this?

See e.g. Node:
```
void Node::get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const {
        const String pf = p_function;
        if (p_idx == 0 && (pf == "has_node" || pf == "get_node" || pf == "get_node_or_null")) {
                _add_nodes_to_options(this, this, r_options);
        } else if (p_idx == 0 && (pf == "add_to_group" || pf == "remove_from_group" || pf == "is_in_group")) {
                HashMap<StringName, String> global_groups = ProjectSettings::get_singleton()->get_global_groups_list();
                for (const KeyValue<StringName, String> &E : global_groups) {
                        r_options->push_back(E.key.operator String().quote());
                }
        }
        Object::get_argument_options(p_function, p_idx, r_options);
}
```

I understand the point about this only matching a single function, but IMO the logic should then be:
- Check the most specific types first for whether they provide arguments for this function
- If yes, return
- Otherwise, try the parent class

So the order should still be swapped here, but all other implementations may need a similar `if (!r_options->is_empty()) { return; }` added.

Reply:
IMO the case for handling the derived class first is that a derived class might have access to data which allows better completion of a parent class type, than the parent class has. This might be particularly relevant for virtual methods whose implementation (and thus the argument values they might work with) can change in derived classes to become more specific.

I don't have a concrete example in mind, and we can very well wait to find one instead of future proofing. But this PR still adds inconsistency with what we do everywhere else, which seems weird to me.